rand_distr/src/utils.rs

```diff
@@ -10,8 +10,50 @@
 
 use rand::Rng;
 use crate::ziggurat_tables;
-
 use rand::distributions::hidden_export::IntoFloat;
+use core::{cmp, ops};
+
+/// Trait for floating-point scalar types
+/// 
+/// This allows many distributions to work with `f32` or `f64` parameters and is
+/// potentially extensible. Note however that the `Exp1` and `StandardNormal`
+/// distributions are implemented exclusively for `f32` and `f64`.
+/// 
+/// The bounds and methods are based purely on internal
+/// requirements, and will change as needed.
+pub trait Float: Copy + Sized + cmp::PartialOrd
```

Comment:
Maybe `pub(crate)` is more appropriate?

Reply:
Then it would be a private trait used in a public API — however, the compiler doesn't complain.

Reply:
Actually, I do get a compiler error if marking this `pub(crate)`, though not if it is a `pub` item within a private module which is not re-exported (I guess this must be a compiler bug and don't know that we should exploit this).

> error[E0445]: crate-visible trait `utils::Float` in public interface

There are discussions on roughly this [here](https://github.com/rust-lang/rust/issues/34537) and [here](https://github.com/rust-lang/rust/issues/34537), but I'm not sure exactly where this got to.